packages/swap/src/routes/v2/swap.ts

```diff
@@ -0,0 +1,408 @@
+    };
+
+    const depositTransactionRef =
+      swapRequest?.depositTransactionRef ??
+      pendingDeposit?.transactionHash ??
+      failedSwap?.depositTransactionRef ??
+      undefined;
+
+    const sortedSwaps = swapRequest?.swaps.sort((a, b) => {
+      if (a.swapExecutedAt && b.swapExecutedAt) {
+        return a.swapExecutedAt.valueOf() - b.swapExecutedAt.valueOf() ? 1 : -1;
+      }
+      return a.swapExecutedAt && !b.swapExecutedAt ? 1 : 0;
+    });
+    // const fees = (swapRequest?.fees ?? []).concat(swap?.fees ?? []);
+    // type: swap?.type,
+
+    const response = {
+      state,
+      ...(internalSrcAsset && getAssetAndChain(internalSrcAsset, 'src')),
+      ...(internalDestAsset && getAssetAndChain(internalDestAsset, 'dest')),
+      depositChannel: {
+        destAddress: readField(swapRequest, swapDepositChannel, failedSwap, 'destAddress'),
+        createdAt: swapDepositChannel?.createdAt.valueOf(),
+        brokerCommissionBps: swapDepositChannel?.brokerCommissionBps,
+        depositAddress: swapDepositChannel?.depositAddress,
+        expectedDepositAmount: swapDepositChannel?.expectedDepositAmount?.toFixed(),
+        expiryBlock: swapDepositChannel?.srcChainExpiryBlock?.toString(),
+        estimatedExpiryTime: swapDepositChannel?.estimatedExpiryAt?.valueOf(),
+        isExpired: swapDepositChannel?.isExpired,
+        openedThroughBackend: swapDepositChannel?.openedThroughBackend,
+        affiliateBrokers,
+        fillOrKillParams: swapDepositChannel?.fokMinPriceX128
+          ? {
+              retryDurationBlocks: swapDepositChannel.fokRetryDurationBlocks,
+              refundAddress: swapDepositChannel.fokRefundAddress,
+              minPrice: getPriceFromPriceX128(
+                swapDepositChannel.fokMinPriceX128.toFixed(),
+                swapDepositChannel.srcAsset,
+                swapDepositChannel.destAsset,
+              ),
+            }
+          : undefined,
+        dcaParams: swapDepositChannel?.chunkIntervalBlocks
+          ? {
+              numberOfChunks: swapDepositChannel?.numberOfChunks,
+              chunkIntervalBlocks: swapDepositChannel?.chunkIntervalBlocks,
+            }
+          : undefined,
+      },
+      deposit: {
+        amount:
+          readField(swapRequest, failedSwap, 'depositAmount')?.toFixed() ?? pendingDeposit?.amount,
+        transactionRef: depositTransactionRef,
+        txConfirmations: pendingDeposit?.transactionConfirmations,
+        receivedAt: swapRequest?.depositReceivedAt?.valueOf(),
+        receivedBlockIndex: swapRequest?.depositReceivedBlockIndex ?? undefined,
+      },
+      swap: {
+        ...sortedSwaps?.reduce(
+          (acc, curr) => {
+            if (isEgressableSwap(curr)) {
+              acc.totalAmountSwapped = acc.totalAmountSwapped.plus(curr.swapOutputAmount ?? 0);
+              if (curr.swapScheduledAt && curr.swapExecutedAt) {
+                acc.lastExecutedChunk = getSwapFields(curr);
```

Comment:
would probably do this and then transform only after reduce has finished
```suggestion
                acc.lastExecutedChunk = curr;
```